### ISSUE TITLE
Errors encountered building Tika-Master from the source

### ISSUE THREAD

**Issue by ipsmile** [deeplearning4j/deeplearning4j#8339]
ubuntu 18.04; java: openjdk version "1.8.0_222"; maven: 3.6.0

The source codes are located at:  https://github.com/apache/tika/archive/master.zip

mvn clean install stopped due to the following error. Just like to report it based on the generated messages. 

If this doesn't help any code improvement, please disregard it.

--------------------------------------------------------------------------------------------------------------
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.tika.dl.imagerec.DL4JVGG16NetTest
log4j:WARN No appenders could be found for logger (org.nd4j.linalg.factory.Nd4jBackend).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.673 s <<< FAILURE! - in org.apache.tika.dl.imagerec.DL4JVGG16NetTest
[ERROR] recognise(org.apache.tika.dl.imagerec.DL4JVGG16NetTest)  Time elapsed: 1.626 s  <<< ERROR!
org.apache.tika.exception.TikaConfigException: Pretrained model file failed checksum. If this error persists, please open an issue at https://github.com/deeplearning4j/deeplearning4j.
	at org.apache.tika.dl.imagerec.DL4JVGG16NetTest.recognise(DL4JVGG16NetTest.java:36)
Caused by: java.lang.IllegalStateException: Pretrained model file failed checksum. If this error persists, please open an issue at https://github.com/deeplearning4j/deeplearning4j.
	at org.apache.tika.dl.imagerec.DL4JVGG16NetTest.recognise(DL4JVGG16NetTest.java:36
[INFO] Running org.apache.tika.dl.imagerec.DL4JInceptionV3NetTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.661 s - in org.apache.tika.dl.imagerec.DL4JInceptionV3NetTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   DL4JVGG16NetTest.recognise:36 » TikaConfig Pretrained model file failed checks...
[INFO] 
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Tika 2.0.0-SNAPSHOT:
[INFO] 
[INFO] Apache Tika parent ................................. SUCCESS [  0.877 s]
[INFO] Apache Tika core ................................... SUCCESS [ 21.913 s]
[INFO] Apache Tika parsers ................................ SUCCESS [01:47 min]
[INFO] Apache Tika OSGi bundle ............................ SUCCESS [ 18.366 s]
[INFO] Apache Tika XMP .................................... SUCCESS [  1.726 s]
[INFO] Apache Tika serialization .......................... SUCCESS [  1.560 s]
[INFO] Apache Tika batch .................................. SUCCESS [02:40 min]
[INFO] Apache Tika language detection ..................... SUCCESS [  2.054 s]
[INFO] Apache Tika application ............................ SUCCESS [01:14 min]
[INFO] Apache Tika translate .............................. SUCCESS [  2.041 s]
[INFO] Apache Tika server ................................. SUCCESS [ 52.455 s]
[INFO] Apache Tika eval ................................... SUCCESS [ 27.267 s]
[INFO] Apache Tika examples ............................... SUCCESS [  6.300 s]
[INFO] Apache Tika Java-7 Components ...................... SUCCESS [  1.960 s]
[INFO] Apache Tika Deep Learning (powered by DL4J) ........ FAILURE [  7.580 s]
[INFO] Apache Tika Natural Language Processing ............ SKIPPED
[INFO] Apache Tika ........................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:06 min
[INFO] Finished at: 2019-11-01T15:37:02-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project tika-dl: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/tssoon/Desktop/tika-master/tika-dl/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :tika-dl

--------------------------------------------------

[TEST-org.apache.tika.dl.imagerec.DL4JInceptionV3NetTest.xml.txt](https://github.com/eclipse/deeplearning4j/files/3799967/TEST-org.apache.tika.dl.imagerec.DL4JInceptionV3NetTest.xml.txt)

**Comment by AlexDBlack** [comment]
Looks like Tika is using DL4J internally, but it's not clear exactly what they are doing (DL4J there is burried behind some layers of Tika APIs)
https://github.com/apache/tika/blob/master/tika-dl/src/test/java/org/apache/tika/dl/imagerec/DL4JVGG16NetTest.java

We would need this to be isolated before we can help (we have model zoo tests and they are all passing).

Alternatively this could have been an intermittent downloading issue - i.e., corrupted download. Have you tried running this multiple times?

**Comment by MissppiBrennan** [comment]
I have this same issue, having run the program multiple times and always breaking at the same point.

**Comment by solomax** [comment]
Just hit this issue as well

Partial log:

```
[INFO] Running org.apache.tika.dl.imagerec.DL4JVGG16NetTest
WARN  [main] 10:24:07,765 org.apache.tika.dl.imagerec.DL4JVGG16Net Preprocessed Model doesn't exist at /home/solomax/.tika-dl/models/dl4j/vgg-16/vgg16.zip
INFO  [main] 10:24:07,768 org.deeplearning4j.zoo.ZooModel Using cached model at /home/solomax/.deeplearning4j/models/vgg16/vgg16_dl4j_inference.zip
INFO  [main] 10:24:07,768 org.deeplearning4j.zoo.ZooModel Verifying download...
INFO  [main] 10:24:07,842 org.deeplearning4j.zoo.ZooModel Checksum local is 2382895241, expecting 3501732770
ERROR [main] 10:24:07,843 org.deeplearning4j.zoo.ZooModel Checksums do not match. Cleaning up files and failing...
WARN  [main] 10:24:07,861 org.apache.tika.dl.imagerec.DL4JVGG16Net Pretrained model file failed checksum. If this error persists, please open an issue at https://github.com/eclipse/deeplearning4j.
java.lang.IllegalStateException: Pretrained model file failed checksum. If this error persists, please open an issue at https://github.com/eclipse/deeplearning4j.
	at org.deeplearning4j.zoo.ZooModel.initPretrained(ZooModel.java:92) ~[deeplearning4j-zoo-1.0.0-M2.1.jar:?]
	at org.apache.tika.dl.imagerec.DL4JVGG16Net.initialize(DL4JVGG16Net.java:107) ~[classes/:?]
	at org.apache.tika.parser.recognition.ObjectRecognitionParser.initialize(ObjectRecognitionParser.java:87) ~[tika-parser-advancedmedia-module-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.apache.tika.config.TikaConfig$XmlLoader.loadOne(TikaConfig.java:767) ~[tika-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.apache.tika.config.TikaConfig$XmlLoader.loadOverall(TikaConfig.java:649) ~[tika-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.apache.tika.config.TikaConfig.<init>(TikaConfig.java:171) ~[tika-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.apache.tika.config.TikaConfig.<init>(TikaConfig.java:151) ~[tika-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
```

The error happens here: https://github.com/apache/tika/blob/main/tika-parsers/tika-parsers-ml/tika-dl/src/main/java/org/apache/tika/dl/imagerec/DL4JVGG16Net.java#L107

Not sure why checksum fails :(

```
OS: Ubuntu 22.04

Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: /opt/maven
Java version: 17.0.8.1, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-83-generic", arch: "amd64", family: "unix"
```

Works as expected in case Java-11 is used: `11.0.20.1`

**Comment by agibsonccc** [comment]
@solomax thanks for flagging. Let me take a look. I'm due to have an M3 release out soon pending some GPU code refactoring. Your team should be able to depend on that after this. Note that the deeplearning4j-zoo module will continue to exist but I would like to work with your team to maybe change some of the code in tika to work with our newer model import functionality and newer modules.

**Comment by solomax** [comment]
Thanks for the quick answer @agibsonccc :)

Adding @tballison who is active Tika maintainer :)

**Comment by treo** [comment]
As a reference, that part of Tika still depends on dl4j beta6 which was released in 2019, so an update to M3 would be a good idea any way.

As for the reason this may fail for you: I guess that it is because the file download failed in a previous run and now you have a half-downloaded file there. Remove the cached file and try again.

**Comment by solomax** [comment]
Hello @treo,

Tika is updating :)

Here https://github.com/apache/tika/tree/TIKA-3948 latest available `m2.1` is used

And the error is reproducible :(

**Comment by treo** [comment]
It is reproducible with a redownload? As far as I can tell, the original file (https://dl4jdata.blob.core.windows.net/models/vgg16_dl4j_inference.zip) does still have the expected checksum (3501732770). 

You can also double check if your file has it outside of java with:
`python -c "import zlib; print(zlib.adler32(open('/home/solomax/.deeplearning4j/models/vgg16/vgg16_dl4j_inference.zip', 'rb').read()));"`

**Comment by tballison** [comment]
FWIW, I wasn't seeing this error. I just cleared my ~/.deeplearning4j cache and got a clean build with a fresh download as well.

**Comment by solomax** [comment]
Hello All,

sorry for delay

To double-check I did the following:
 - `rm -rf ~/.tika-dl ~/.deeplearning4j`
 - `MAVEN_OPTS="-Xmx8g" mvn clean install` (at Tika `TIKA-3948` branch)
 - The build is GREEN

So NREP for me as well
(not sure what was it :((( )


sorry for the noise

**Comment by THausherr** [comment]
https://dl4jdata.blob.core.windows.net/models/vgg16_dl4j_inference.zip

"The specified account is disabled"

Is there a new URL for this? We are using 1.0.0-M2.1.

**Comment by agibsonccc** [comment]
@THausherr  let me look at that. I've paid the bill on that azure account for now but I will be migrating to github as the CDN of choice instead.

**Comment by timer-180** [comment]
https://dl4jdata.blob.core.windows.net/
It's sill not working - AccountIsDisabled error code.
Is there any updates on this?

**Comment by agibsonccc** [comment]
Closing — this issue is over 5 years old with no recent activity. The codebase has been significantly rewritten. If this is still relevant, please open a new issue.
